jfrog-client/artifactory/services/go/go.go

```diff
@@ -51,12 +63,37 @@ func addPropertiesHeaders(props string, headers *map[string]string) error {
 	return nil
 }
 
+func createUrlPath(params GoParams, url *string) error {
+	*url = strings.Join([]string{*url, params.GetModuleId(), "@v", params.GetVersion() + ".zip"}, "/")
+	properties, err := utils.ParseProperties(params.GetProps(), utils.JoinCommas)
+	if err != nil {
+		return err
+	}
+	*url = strings.Join([]string{*url, properties.ToEncodedString()}, ";")
+	if strings.HasSuffix(*url, ";") {
+		tempUrl := *url
+		tempUrl = tempUrl[:len(tempUrl)-1]
+		*url = tempUrl
+	}
+	return nil
+}
+
+// Returns true if needed to use properties as header or false if need to use matrix params.
+func shouldUseHeaders(artifactoryVersion string) bool {
```

Comment:
I suggest adding a unit test for this function.
Also, to make the code a bit more easier to read, I suggest adding to the above function description, the min Artifactory version for which this function returns *true*.

Reply:
Done. I've added a test a changed the documentation.